doc/test_messages_documentation.py

```diff
@@ -146,21 +146,25 @@ def get_expected_messages(stream: TextIO) -> MessageCounter:
                 messages[lineno, msg_id.strip()] += 1
         return messages
 
-    def _get_expected(self) -> MessageCounter:
+    def _get_expected(self) -> dict[Path, MessageCounter]:
         """Get the expected messages for a file or directory."""
-        expected_msgs: MessageCounter = Counter()
+        expected_msgs: dict[Path, MessageCounter] = {}
         if self._test_file[1].is_dir():
-            for test_file in self._test_file[1].iterdir():
+            for _i, test_file in enumerate(self._test_file[1].iterdir()):
+                expected_msgs[test_file] = Counter()
                 with open(test_file, encoding="utf8") as f:
-                    expected_msgs += self.get_expected_messages(f)
+                    expected_msgs[test_file] += self.get_expected_messages(f)
         else:
+            expected_msgs[self._test_file[1]] = Counter()
             with open(self._test_file[1], encoding="utf8") as f:
-                expected_msgs += self.get_expected_messages(f)
+                expected_msgs[self._test_file[1]] += self.get_expected_messages(f)
         return expected_msgs
 
     def _get_actual(self) -> MessageCounter:
         """Get the actual messages after a run."""
         messages: list[Message] = self._linter.reporter.messages
+        str_messages = "\n- ".join([str(m) for m in messages])
+        print(f"Actual messages raised:\n- {str_messages}")
```

Comment:
Debug?

Reply:
Yes it's helpful to have the info in the log in the CI. Actually it also contain the info about which file has a message raised (which would help make a better error message) but doing thing properly would make us almost redo everything from scratch so giving the info with a print felt adequate.
```
Config file used: None

Command used:

pylint /home/pierre/pylint/doc/data/messages/c/cell-var-from-loop/good --disable=all --enable=F,cell-var-from-loop,astroid-error,syntax-error

Actual messages raised:

- Message(msg_id='W0640', symbol='cell-var-from-loop', msg='Cell variable name defined in loop', C='W', category='warning', confidence=Confidence(name='UNDEFINED', description='Warning without any associated confidence level.'), abspath='/home/pierre/pylint/doc/data/messages/c/cell-var-from-loop/good/functools.partial.py', path='data/messages/c/cell-var-from-loop/good/functools.partial.py', module='functools.partial', obj='teacher_greeting.greet', line=8, column=66, end_line=8, end_column=70)
```

Reply:
Hmm I'm not a big fan of leaving such print statements in code (there is a `pylint` warning for it :smile:) as you don't have any control over the degree of logging.

But since we don't touch this code often I'll leave this decision to you.